lib/spack/docs/index.rst

```diff
@@ -37,21 +37,30 @@ package:
 If you're new to spack and want to start using it, see :doc:`getting_started`,
 or refer to the full manual below.
 
------------------
-Table of Contents
------------------
+--------------------------
+Documentation and Tutorial
+--------------------------
 
 .. toctree::
    :maxdepth: 2
 
    features
    getting_started
    basic_usage
+   spack_workflows
+
+
+----------------
+Reference Manual
+----------------
+
+.. toctree::
+   :maxdepth: 2
```

Comment:
I delved a little further into this and the way you're actually supposed to do it is:
```rst
.. toctree::
   :maxdepth: 2
   :caption: Documentation and Tutorial

   features
   getting_started
   basic_usage
   spack_workflows

.. toctree::
   :maxdepth: 2
   :caption: Reference Manual

   packaging_guide
   ...
```
That's how [ReadTheDocs documentation](http://read-the-docs.readthedocs.io/en/latest/index.html) gets that nice division on the left panel. You can see their source code here:
https://raw.githubusercontent.com/rtfd/readthedocs.org/master/docs/index.rst